pkg/provision/workspace/deployment.go

```diff
@@ -300,6 +315,23 @@ func getSpecDeployment(
 	return deployment, nil
 }
 
+// Returns the ProgressDeadlineSeconds to use for workspace deployments as an int32.
+// The ProgressDeadLineSeconds returned is the same length of time as DWOC's workspace.ProgressTimeout.
+// Returns an error if ProgressTimeout could not be properly parsed,
+// or if the ProgressTimeout would exceed the maximum capacity of an int32
+func getProgressDeadlineSeconds(config *v1alpha1.OperatorConfiguration) (int32, error) {
+	timeout, err := time.ParseDuration(config.Workspace.ProgressTimeout)
+	if err != nil {
+		return 0, fmt.Errorf("invalid duration specified for workspace progress timeout: %w", err)
+	}
+	timeoutRounded := timeout.Round(time.Second)
```

Comment:
Do you need to round? I believe converting to an `int32` will drop the factional part.

Reply:
You're right, no need to round -- I was just being overly cautious :P